staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +49,17 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+	// GetAnnotations returns a copy of saved annotations. Modification to the retruned copy of annotations will not be saved to Attributes. Use SetAnnotation/SetAnnotations
+	// to set annotations of Attributes.
+	GetAnnotations() map[string]string
+	// SetAnnotation sets annotation according to key-value pair. The key should be qualified, e.g., podsecuritypolicy.admission.k8s.io/admit-policy, where
+	// "podsecuritypolicy" is the name of the plugin, "admission.k8s.io" is the name of the organization, "admit-policy" is the key name.
+	// Both ValidationInterface and MutationInterface are allowed to set Annotations.
+	SetAnnotation(key, value string)
+	// SetAnnotations sets annotations according to key-value pairs in annotations map. The keys of annotations map should be qualified, e.g., podsecuritypolicy.admission.k8s.io/admit-policy,
+	// where "podsecuritypolicy" is the name of the plugin, "admission.k8s.io" is the name of the organization, "admit-policy" is the key name.
+	// Both ValidationInterface and MutationInterface are allowed to set Annotations.
+	SetAnnotations(annotations map[string]string)
```

Comment:
I'm still not sold that mutable attributes is the right interface for this. What if instead we modify the admission interfaces to return a result struct that (optionally) includes annotations?

```go
Admit(a Attributes) (Result, error)
Validate(a Attributes) (Result, error)

type Result struct {
  Decision Decision
  Reason string  // Do we care about this?
  AuditInfo map[string]string
  // Optional: HandlerName string
}

type Decision string
const (
  DecisionAccept Decision = "Accept"
  DecisionReject Decision = "Reject"
)
```

@sttts @liggitt WDYT?

Reply:
Waiting for @sttts and @liggitt s' suggestions.

Reply:
Note that the new object returned via `GetObject()` is also mutable. 

Also note that the synchronization of those mutations is not implicit by the attributes implementation, but done by the callee. With the annotation funcs we differ from those semantics.

Reply:
Updated the comment.

Reply:
@sttts Are you suggesting that since the object is mutable it's reasonable to mutate the annotations on this interface?

One difference is that the object is only considered mutable during the mutating admission phase, while the annotations can also be modified during the validating phase.

Reply:
@tallclair was just an observation, am not suggesting that this is a good interface. In general, let's keep the interface small (only `AddAnnotation`, not getter). But at that point we are not far from your `Result` type which I would also like.

Reply:
Thanks very for your suggestions will work on this soon.

Reply:
> In general, let's keep the interface small (only AddAnnotation, not getter). 

Done.

@tallclair @sttts Please take a look.
This version only adds one function to the interface.